lib/bitcoin/lock/LockMonitor.ts

```diff
@@ -127,6 +172,8 @@ export default class LockMonitor {
     if (!(await this.isTransactionWrittenOnBitcoin(lastSavedLock.transactionId))) {
 
       await this.rebroadcastTransaction(lastSavedLock);
+      currentLockState.status = LockStatus.Pending;
+
       return currentLockState;

```

Comment:
Consider constructing and return the full state object within these short-circuit return blocks for readability.

Reply:
Updated.